integrations/oci/metrics/metrics/src/main/java/io/helidon/integrations/oci/metrics/OciMetricsSupport.java

```diff
@@ -0,0 +1,424 @@
+        public Builder nameFormatter(NameFormatter nameFormatter) {
+            this.nameFormatter = nameFormatter;
+            return this;
+        }
+
+        /**
+         * Sets the resource group.
+         *
+         * @param value resource group
+         * @return updated builder
+         */
+        @ConfiguredOption
+        public Builder resourceGroup(String value) {
+            resourceGroup = value;
+            return this;
+        }
+
+        /**
+         * Sets whether the description should be enabled or not.
+         *
+         * @param value enabled
+         * @return updated builder
+         */
+        @ConfiguredOption
+        public Builder descriptionEnabled(boolean value) {
+            descriptionEnabled = value;
+            return this;
+        }
+
+        /**
+         * Sets which metrics scopes (e.g., base, vendor, application) should be sent to OCI.
+         * <p>
+         *     If this method is never invoked, defaults to all scopes.
+         * </p>
+         *
+         * @param value list of metric scopes to process
+         * @return updated builder
+         */
+        @ConfiguredOption
+        public Builder scopes(List<String> value) {
+            Map<String, Type> scopeTypes = Map.of(
+                    "base", Type.BASE,
+                    "vendor", Type.VENDOR,
+                    "application", Type.APPLICATION
+            );
+            if (value == null || value.size() == 0) {
+                this.scopes =  new ArrayList<Type>(scopeTypes.values()).toArray(new Type[scopeTypes.size()]);
+            } else {
+                List<Type> convertedScope = new ArrayList<>();
+                for (String element: value) {
+                    Type scopeItem = scopeTypes.get(element);
+                    if (scopeItem != null) {
+                        convertedScope.add(scopeItem);
+                    }
+                }
+                this.scopes = convertedScope.toArray(new Type[convertedScope.size()]);
+            }
+            return this;
+        }
+
+        /**
+         * Sets whether metrics transmission to OCI is enabled.
+         * <p>
+         *     Defaults to {@code true}.
+         * </p>
+         * @param value whether metrics transmission should be enabled
+         * @return updated builder
+         */
+        @ConfiguredOption
+        public Builder enabled(boolean value) {
+            enabled = value;
+            return this;
+        }
+
+        /**
+         * Updates the builder using the specified OCI metrics {@link Config} node.
+         * @param config {@code Config} node containing the OCI metrics settings
+         * @return updated builder
+         */
+        public Builder config(Config config) {
+            config.get("initialDelay").asLong().ifPresent(this::initialDelay);
+            config.get("delay").asLong().ifPresent(this::delay);
+            config.get("compartmentId").asString().ifPresent(this::compartmentId);
```

Comment:
We should allow config to specify the time units for the scheduling. I'll send some code and tests separately.

Reply:
Implemented in the most recent commit.